src/main/kotlin/io/sip3/salto/ce/socket/ManagmentSocket.kt

```diff
@@ -0,0 +1,156 @@
+
+    private val remoteHosts = mutableMapOf<String, RemoteHost>()
+    lateinit var socket: DatagramSocket
+
+    override fun start() {
+        config().getJsonObject("mongo")?.let { config ->
+            client = MongoClient.createShared(vertx, JsonObject().apply {
+                put("connection_string", config.getString("uri") ?: throw IllegalArgumentException("mongo.uri"))
+                put("db_name", config.getString("db") ?: throw IllegalArgumentException("mongo.db"))
+            })
+        }
+
+        config().getJsonObject("management").let { config ->
+            uri = URI(config.getString("uri") ?: throw IllegalArgumentException("uri"))
+            config.getLong("expiration-timeout")?.let { expirationTimeout = it }
+            config.getLong("expiration-delay")?.let { expirationDelay = it }
+        }
+
+        startUdpServer()
+
+        vertx.eventBus().localConsumer<List<SdpSession>>(RoutesCE.sdp_session) { event ->
+            val sdpSessions = event.body()
+            sdpSessions.forEach { publishSdpSession(it) }
+        }
+
+        vertx.setPeriodic(expirationDelay) {
+            val now = System.currentTimeMillis()
+            remoteHosts.filterValues { it.lastUpdate + expirationTimeout < now }
+                    .forEach { remoteHosts.remove(it.key) }
+        }
+    }
+
+    private fun publishSdpSession(sdpSession: SdpSession) {
+        val buffer = JsonObject().apply {
+            put("type", TYPE_SDP_SESSION)
+            put("payload", JsonObject.mapFrom(sdpSession))
+        }.toBuffer()
+
+        remoteHosts.values.forEach { remoteHost ->
+            try {
+                socket.send(buffer, remoteHost.uri.port, remoteHost.uri.host) {}
+            } catch (e: Exception) {
+                logger.error("Failed to send to ${remoteHost.uri}: ${e.message}", e)
+            }
+        }
+    }
+
+    private fun startUdpServer() {
+        socket = vertx.createDatagramSocket()
+        socket.handler { packet ->
+            val buffer = packet.data()
+            val socketAddress = packet.sender()
+            try {
+                val message = buffer.toJsonObject()
+                handle(message, socketAddress)
+            } catch (e: Exception) {
+                logger.error("ManagementSocket 'handle()' failed.", e)
+            }
+        }
+
+        socket.listen(uri.port, uri.host) { connection ->
+            if (connection.failed()) {
+                logger.error("UDP connection failed. URI: $uri", connection.cause())
+                throw connection.cause()
+            }
+            logger.info("Listening on $uri")
+        }
+    }
+
+    private fun handle(message: JsonObject, socketAddress: SocketAddress) {
+        val type = message.getString("type")
+        val payload = message.getJsonObject("payload")
+
+        when (type) {
+            TYPE_REGISTER -> {
+                val name = payload.getString("name")
+
+                remoteHosts.computeIfAbsent(name) {
```

Comment:
I suggest to add `lastUpdate` as a property in data class. So here we can just assign the value to map instead of updating `lastUpdate` only. Also `payload.getJsonObject("host")?.let { updateHost(it) }` looks confusing inside apply...

Reply:
I agree with suggestion about `updateHost()` placement. 
But if we just assign `RemoteHost` on each message, we need to recalc and recreate URI for this host every time. Not sure we need it because of rare starts of remote hosts.